doc/sag/sag-HLD.md

```diff
@@ -0,0 +1,185 @@
+It should always to configure VxLAN to make SAG work.  
+SAG announces its IP and MAC only towards the host facing Ethernet ports but not towards the fabric (Overlay).  
+As a result, the IP/MAC for the SAG is only seen and reachable from the host facing side and never from the fabric (Overlay) facing side.
+
+The following is the requirements:  
+* Support one global virtual MAC address on SAG interface
+* Support multiple IPv4/IPv6 address on SAG interface
+* Support global IPv4/IPv6 enable knob for SAG address
+
+# Modules Design
+## Config DB
+### SAG Global Table
+```
+SAG_GLOBAL|IP 
+  "gwmac": {{mac_address}} 
+  "IPv4": {{enable}}
+  "IPv6": {{enable}}
+```
+### SAG Interface Table
+```
+SAG|{{vlan_intf_name}}|IPv4
+  "gwip@": {{ipv4_list}}
+SAG|{{vlan_intf_name}}|IPv6
+  "gwip@": {{ipv6_ilst}}
+```
+
+### Config DB Schemas
+```
+; Defines schema for SAG configuration attributes
+key                                   = SAG_GLOBAL|IP                 ; SAG global configuration
+; field                               = value
+gwmac                                 = mac_address                   ; global anycast gateway MAC address
+IPv4                                  = enable/disable                ; enable anycast gateway IPv4 address or not
+IPv6                                  = enable/disable                ; enable anycast gateway IPv6 address or not
+	  
+```
+## APP DB
+### Interface Table
+```
+INTF_TABLE:{{intf_name}}
+  "mac_addr": {{mac_address}}
+
+INTF_TABLE:{{intf_name}}:{{ip_prefix}}
+  "scope": {{addr_scope}
+  "family": {{ip_family}}
+```
+### APP DB Schemas
+```
+; Defines schema for SAG configuration attributes
+key                                   = INTF_TABLE:sag_name           ; SAG interface 
+; field                               = value
+mac_addr                              = mac_address                   ; global anycast mac MAC address
+
+key                                   = INTF_TABLE:sag_name:ip_prefix ; SAG IP address
+; field                               = value
+scope                                 = global/local/host             ; IP address scope
+family                                = IPv4/IPv6                     ; IP address family
+```
+
+## Orchestration Agent
+Following orchagents shall be modified. Flow diagrams are captured in a later section. 
+
+![SAG arch](./images/sag-db-arch.png)
+
+### SagMgrD
+SagMgrD creates the macvlan kernel interfaces on top of the vlan interface.  
+SagMgrD waits for macvlan interface creation update in STATE_DB and updates the APP_DB INTF_TABLE with the macvlan interface name.
+SagMgrD creates IP address on the SAG interface. 
+SagMgrD gets the INTERFACE Table config to create route entry in SAI from APP_DB updates
+
+### IntfsOrch
+Add SagOrch as a member of IntfsOrch. IntfsOrch creates Router Interfaces based on interface table (INTF_TABLE). For SAG usecase, IntfOrch calls SagOrch API to handle router interface creation.
```

Comment:
Can we handle this within IntfsOrch instead of creating a new SagOrch?

Reply:
Use IntfsOrch to handle the SAG directly instead of creating a new Orch in the updated HLD.